### PR TITLE
Python ScriptEngine:: Step 3: Get back the results & Step 4: Get back the logs

### PR DESCRIPTION
**Pull request by sophiesongge**

### CODE REVIEW

File: src/main/java/jsr223/cpython/python/PythonScriptWriter.java [hunk 1]
```diff
@@ -55,10 +60,29 @@ public File writeFileToDisk(String fileContent) throws IOException {
...
+            writeLine("if '" + FlowScript.loopVariable + "' in locals():", pythonScriptBufferedWriter);
+            writeLine("    bindings['" + FlowScript.loopVariable + "'] = " + FlowScript.loopVariable + "",
+                      pythonScriptBufferedWriter);
+            writeLine("if '" + SchedulerConstants.RESULT_METADATA_VARIABLE + "' in locals():",
+                      pythonScriptBufferedWriter);
+            writeLine("    bindings['" + SchedulerConstants.RESULT_METADATA_VARIABLE + "'] = " +
+                      SchedulerConstants.RESULT_METADATA_VARIABLE + "", pythonScriptBufferedWriter);
         } catch (IOException e) {
-            throw new IOException("Unable to write the python scripts to a file. " + e);
+            throw new IOException("Unable to write the python scripts to a file. " + e.getMessage());
```
Comment (fviale) [inline src/main/java/jsr223/cpython/python/PythonScriptWriter.java]: Better this:

throw new IOException("Unable to write the python scripts to a file.",  e);

**Review by fviale — APPROVED** [review]

File: src/main/java/jsr223/cpython/entrypoint/EntryPoint.java [hunk 2]
```diff
@@ -50,14 +50,14 @@ public static EntryPoint getInstance() {
...
     }
 
-    private Map<String, Serializable> variabels = new HashMap<>();
+    private Map<String, Serializable> variables = new HashMap<>();
 
-    public Bindings bindings = new SimpleBindings();
+    private Bindings bindings = new SimpleBindings();
 
     //Add the objects to the gateway server
     //TODO not sure we still need this getVariables() because we already have the getBindings() whtich contains all
```
Comment (tobwiens) [inline src/main/java/jsr223/cpython/entrypoint/EntryPoint.java:58]: Maybe you can remove this todo.

File: src/main/java/jsr223/cpython/entrypoint/EntryPoint.java [hunk 3]
```diff
@@ -50,14 +50,14 @@ public static EntryPoint getInstance() {
...
     public Map getVariables() {
-        return variabels;
+        return variables;
```
Comment (tobwiens) [inline src/main/java/jsr223/cpython/entrypoint/EntryPoint.java:60]: Question to think about: Do you want to give access to the variables, or do you want to protect the variables and return a copy of the map?

**Review by tobwiens — APPROVED** [review]
Looks good.

A few optional improvements.

File: src/main/java/jsr223/cpython/entrypoint/EntryPoint.java (hunk 3 above)
Comment (sophiesongge) [reply src/main/java/jsr223/cpython/entrypoint/EntryPoint.java:60]: Hi, I want to give access to the variables here :)

**Pull request merged**